crates/query-engine/translation/src/translation/mutation/v2/delete.rs

```diff
@@ -11,22 +11,26 @@ use query_engine_metadata::metadata::database;
 use query_engine_sql::sql;
 use std::collections::BTreeMap;
 
-use super::common::{self, CheckArgument};
+use super::common::{self, default_constraint, CheckArgument};
 
 /// A representation of an auto-generated delete mutation.
 ///
 /// This can get us `DELETE FROM <table> WHERE column = <column_name_arg>, ...`.
 #[derive(Debug, Clone)]
 pub enum DeleteMutation {
-    DeleteByKey {
-        description: String,
-        collection_name: models::CollectionName,
-        schema_name: sql::ast::SchemaName,
-        table_name: sql::ast::TableName,
-        by_columns: NonEmpty<metadata::database::ColumnInfo>,
-        columns_prefix: String,
-        pre_check: CheckArgument,
-    },
+    DeleteByKey(DeleteByKey),
+}
+
+/// A representation of an auto-generated delete mutation by a unique key.
+#[derive(Debug, Clone)]
+pub struct DeleteByKey {
+    pub description: String,
+    pub collection_name: models::CollectionName,
+    pub schema_name: sql::ast::SchemaName,
+    pub table_name: sql::ast::TableName,
+    pub by_columns: NonEmpty<metadata::database::ColumnInfo>,
+    pub columns_prefix: String,
+    pub pre_check: Option<CheckArgument>,
```

Comment:
The `Option` here isn't necessary, this can be `CheckArgument`.

Reply:
(it's optional that the user provides the argument, but the argument will always be in the schema, just nullable)